databricks/koalas/tests/test_stats.py

```diff
@@ -122,3 +122,53 @@ def test_stats_on_boolean_series(self):
 
         self.assertAlmostEqual(ds.var(), s.var())
         self.assertAlmostEqual(ds.std(), s.std())
+
+    def test_some_stats_functions_should_discard_non_numeric_columns_by_default(self):
+        df = pd.DataFrame({'i': [0, 1, 2],
+                           'b': [False, False, True],
+                           's': ['x', 'y', 'z']})
+        ddf = koalas.from_pandas(df)
+
+        # min and max do not discard non-numeric columns by default
+        self.assertEqual(len(ddf.min()), 3)
```

Comment:
Regarding all `assertEqual` statements: Maybe it makes more sense to compare the results from the Koalas aggregations with pandas rather than the actual results, such that line 133 could for example be replaced with `self.assertEqual(len(kdf.min()), len(pdf.min()))`. This focuses more on the consistency between both libraries while I believe pandas is pretty well-tested itself so we don't need to write all sorts of tests twice. Maybe @HyukjinKwon also has an opinion about this?

Reply:
Fixing this led me to realise that koalas behavior differs from pandas since in koalas we can apply stats function on non-numeric columns; the returned value is just `NaN`. In pandas, forcing to compute the `mean` on a string column raises an `TypeError`. Do we want to keep pandas behavior, and raise an error ?
Personaly, I'd favor the simplest path for now, and would create another issue for this change later on, if needed. Any strong opinion about this ?

Reply:
Yea I agree with comparing to Pandas' in general. Handling NaN is pretty annoying problem currently in Koalas.

> forcing to compute the mean on a string column raises

It would be better to match it with Pandas' for now. Creating another issue makes sense to me. We can fix it here too. I'll leave it to you @nitlev

Reply:
I also prefer pandas' behavior as it keeps the user from producing unintended results by accident. Maybe it's cleaner to move this to a new PR though and merge this one if you approve @HyukjinKwon.